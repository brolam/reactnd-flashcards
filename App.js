import React from 'react';
import { Text, View } from 'react-native';
import appStyles from './styles/appStyles'
import AppStatusBar from './components/AppStatusBar'
import DeckList from './components/DeckCardList'
import DecksAndQuizzesPanels from './components/DecksAndQuizzesPanels'
import { isPossibleTwoPanels } from './util/ScreenHelper';
import Dimensions from 'Dimensions';

export default class App extends React.Component {
  constructor(props) {
    super();
    this.state = { isTwoPanels: props.isTwoPanels };
  }

  onLayoutChanged = event => {
    this.setState({ isTwoPanels: isPossibleTwoPanels(Dimensions.get('window')) });
  }

  render() {
    return (
      <View style={appStyles.container} onLayout={this.onLayoutChanged} >
        <AppStatusBar />
        {
          (this.state.isTwoPanels) ?
            (
              <DecksAndQuizzesPanels decks={[
                { key: 'one-item', title: 'One Deck', amountOfCards: 10 },
                { key: 'two-item', title: 'Two Deck', amountOfCards: 11 },
                { key: 'three-item', title: 'Three Deck', amountOfCards: 12 }
              ]}
              />
            ) :
            (
              <DeckList decks={[
                { key: 'one-item', title: 'One Deck', amountOfCards: 10 },
                { key: 'two-item', title: 'Two Deck', amountOfCards: 11 },
                { key: 'three-item', title: 'Three Deck', amountOfCards: 12 }
              ]} />
            )
        }
      </View>
    );
  }
}
