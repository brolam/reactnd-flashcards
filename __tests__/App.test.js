import React from 'react';
import App from '../App';

test('renders without crashing', () => {
  const wrapper = shallow(<App isTwoPanels={false} />);
  expect(wrapper).toMatchSnapshot();
});

test('renders two panels without crashing', () => {
  const wrapper = shallow(<App isTwoPanels={true} />);
  expect(wrapper).toMatchSnapshot();
});