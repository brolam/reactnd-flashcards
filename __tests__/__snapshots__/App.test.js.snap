// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders two panels without crashing 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <AppStatusBar />
  <DecksAndQuizzesPanels
    decks={
      Array [
        Object {
          "amountOfCards": 10,
          "key": "one-item",
          "title": "One Deck",
        },
        Object {
          "amountOfCards": 11,
          "key": "two-item",
          "title": "Two Deck",
        },
        Object {
          "amountOfCards": 12,
          "key": "three-item",
          "title": "Three Deck",
        },
      ]
    }
  />
</View>
`;

exports[`renders without crashing 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <AppStatusBar />
  <DeckCardList
    decks={
      Array [
        Object {
          "amountOfCards": 10,
          "key": "one-item",
          "title": "One Deck",
        },
        Object {
          "amountOfCards": 11,
          "key": "two-item",
          "title": "Two Deck",
        },
        Object {
          "amountOfCards": 12,
          "key": "three-item",
          "title": "Three Deck",
        },
      ]
    }
  />
</View>
`;
